homeassistant/components/tedee/sensor.py

```diff
@@ -0,0 +1,74 @@
+"""Tedee sensor entities."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pytedee_async import TedeeLock
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import TedeeDescriptionEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class TedeeSensorEntityDescription(SensorEntityDescription):
+    """Describes Tedee sensor entity."""
+
+    value_fn: Callable[[TedeeLock], float | None]
+
+
+ENTITIES: tuple[TedeeSensorEntityDescription, ...] = (
+    TedeeSensorEntityDescription(
+        key="battery_sensor",
+        device_class=SensorDeviceClass.BATTERY,
```

Comment:
Battery level measurement sensors are normally marked as diagnostic entity category unless the battery is the main feature of the device.